subsys/esb/esb.c

```diff
@@ -766,44 +843,51 @@ static void on_radio_disabled_tx_wait_for_ack(void)
 			 * be entered again as soon as the system timer reaches
 			 * CC[1].
 			 */
-			NRF_RADIO->SHORTS = radio_shorts_common |
-					    RADIO_SHORTS_DISABLED_RXEN_Msk;
+			nrf_radio_shorts_set(NRF_RADIO,
+				(radio_shorts_common | NRF_RADIO_SHORT_DISABLED_RXEN_MASK));
 			update_rf_payload_format(current_payload->length);
-			NRF_RADIO->PACKETPTR = (uint32_t)tx_payload_buffer;
+
+			nrf_radio_packetptr_set(NRF_RADIO, tx_payload_buffer);
+
 			on_radio_disabled = on_radio_disabled_tx;
 			esb_state = ESB_STATE_PTX_TX_ACK;
-			ESB_SYS_TIMER->TASKS_START = 1;
-			nrfx_gppi_channels_enable(1 << ppi_ch_timer_compare1_radio_txen);
-			if (ESB_SYS_TIMER->EVENTS_COMPARE[1]) {
-				NRF_RADIO->TASKS_TXEN = 1;
+
```

Comment:
**NRF_TIMER_TASK_START** should be triggered for **esb_timer** here to make the retransmission work, since **ppi_ch_radio_address_timer_stop** PPI channel will make **esb_timer** stop upon getting ADDRESS event from Radio.

Reply:
Right I forget that we can get packet with invalid CRC